HTTPServer.hpp assumes that chars are signed types. Cannot compile on ARM cpus

HTTPServer.hpp:119:45: error: comparison is always true due to limited range of data type [-Werror=type-limits]
  119 |         auto isChar = [](char c) { return c >= 0; };
      |                                           ~~^~~~
TTPServer.hpp: In lambda function:
HTTPServer.hpp:120:46: error: comparison is always true due to limited range of data type [-Werror=type-limits]
  120 |         auto isCtrl = [](char c) { return (c >= 0 && c <= 31) || (c == 127); };
      |                                            ~~^~~~

--- a/include/http/HTTPServer.hpp
+++ b/include/http/HTTPServer.hpp
@@ -116,8 +116,8 @@ private: // clang-format off
                  versionMinor, newline1, headerLineStart, headerLws, headerName, spaceBeforeHeaderValue, headerValue, newline2, newline3, completed };
     State state { State::methodStart };    
     bool completeRequest(char input) {
-        auto isChar = [](char c) { return c >= 0; };
-        auto isCtrl = [](char c) { return (c >= 0 && c <= 31) || (c == 127); };
+        auto isChar = [](char c) { return static_cast<unsigned char>(c) <= 127; };
+        auto isCtrl = [](char c) { return static_cast<unsigned char>(c) <= 31 || c == 127; };
         auto isSpecial = [](char c) {   switch (c) {
             case '(': case ')': case '<': case '>': case '@': case ',': case ';': case ':': case '\\': case '"':
             case '/': case '[': case ']': case '?': case '=': case '{': case '}': case ' ': case '\t': return true;

